pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -690,6 +690,68 @@ def plot(self, **kwargs):
         if "filename" in kwargs:
             fig.savefig(kwargs["filename"])
 
+
+    def plot2d(self, **kwargs):
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+        lvl = kwargs.get("level", 0)
+        qty = kwargs.get("qty", None)
+        time = kwargs.get("time", self.times()[0])
+        if "ax" not in kwargs:
+            fig, ax = plt.subplots()
+        else:
+            ax = kwargs["ax"]
+            fig = ax.figure
+
+        for patch in self.level(lvl, time).patches:
+            pdat = patch.patch_datas[qty]
+            data = pdat.dataset[:]
+            nbrGhosts = 5 # CLEAN THIS
+            x = pdat.x
+            y = pdat.y
+            nx,ny =x.size, y.size
+            data = data.reshape((nx,ny))
+            data = data[nbrGhosts:-nbrGhosts, nbrGhosts:-nbrGhosts]
+            x = x[nbrGhosts:-nbrGhosts]
+            y = y[nbrGhosts:-nbrGhosts]
+            dx = x[1] - x[0]
+            dy = y[1] - y[0]
+            x -= dx/2
```

Comment:
better use `x -= 0.5*dx` as multiplying is faster

Reply:
not sure it changes anything in python (or in cpp actually)
changed anyway